### DISCUSSION TITLE
How to access defaultValues?

### DISCUSSION THREAD

**Discussion by ersinakinci** [react-hook-form/react-hook-form#2953] [Q&A]
Is there a way to access `defaultValues` using the `useForm`/`useFormContext` hooks?

I'm building a set of JSON document editing controls for a CMS. Every field type gets its own control. For example, a text field gets a string input control. Each control also has a reset button to revert back to its default value as determined by `defaultValues`.

This approach breaks down for one control that creates associations between documents using id's (e.g., foreign key id's). I model the association data in react-hook-form by registering multiple `<input>`'s, one for each association. Each one has a name like `assoc[1]`, `assoc[2]`, etc. As the user adds or removes associations, the control creates and removes inputs. I need some way to be able to reset the entire control to restore the original inputs based on the original id's, which are stored in `defaultValues`. I can't simply do a `reset` because I'm not resetting the inputs themselves, I need to create/destroy several inputs in order to reset the control's state.

I understand that I could achieve my goal in a million ways that don't involve react-hook-form. However, for various reasons, I'm hoping to find a way to do it using the library.

**NOTE**: I just discovered something called `useFieldArray`. Could this somehow help me achieve my aim? I'm not sure if it's applicable to my scenario.

**Comment by bluebill1049** [comment]
you can store your defaultValues in context and access themm.

**Reply by ersinakinci** [reply]
Do you mean within the form context and access them with `useFormContext`? As I mentioned, I'm aware there are a lot of ways to do this outside of react-hook-form, but I was really hoping to do it with just react-hook-form and nothing else.

**Reply by bluebill1049** [reply]
we used to return getValues with defaultValues, this creates a lot of side effect and unexpected result, so prefer for you/user to handle that.
